content/blog/2024/jupyterhub-binderhub-gesis/index.md

```diff
@@ -0,0 +1,126 @@
+
+But, mybinder.org (and the [BinderHub](https://github.com/jupyterhub/binderhub/) software that powers it) is built for *sharing* your work after you are done with it, *not* for actively doing work. [JupyterHub](https://jupyter.org/hub) is more commonly used for this, but doesn't currently have the ability for users to easily build their own environments. Admins who are *running* the JupyterHub can make [multiple environments](https://z2jh.jupyter.org/en/stable/jupyterhub/customizing/user-environment.html#using-multiple-profiles-to-let-users-select-their-environment) available for users to choose from, but this still puts admins in the critical path for environment customization.
+
+Our [collaboration](https://2i2c.org/blog/2022/gesis-2i2c-collaboration-update/) with [GESIS](http://notebooks.gesis.org), [NFDI4DS](https://www.nfdi4datascience.de), and [CESSDA](https://www.cessda.eu), aims to bring this flexibility to JupyterHub directly. We aim to empower users to decide for themselves which applications and dependencies are installed on a per-project basis. Our work enables communities with heterogeneous requirements to share a single Hub. Our approach frees administrators from being overwhelmed by installation requests and transforms the JupyterHub platform into a platform for collaborative computational reproducibility. In this update, we report on our progress and upcoming steps in this project.
+
+## What does a BinderHub do, exactly?
+
+It is helpful to understand that BinderHub primarily has 3 responsibilities:
+
+1. Present a UI to the end user for them to provide details on what to build (this is what you see when you go to mybinder.org)
+2. Call out to [repo2docker](https://github.com/jupyterhub/repo2docker) in a scalable way to actually *build and push* a docker image containing the environment for the given repository, and show the user logs as this build process happens. This also allows users to debug issues with their build more easily.
+3. Talk to a JupyterHub instance to launch a user server with the built docker image, and redirect the user to this.
+
+(2) is really the *core* feature of BinderHub, and we settled on figuring out how to make that available to JupyterHub users. It was really important to us that this was also done in a way that can be sustainably used by *everyone*, not just 2i2c. This blog post discusses the various improvements to the broad ecosystem of projects in the Jupyter ecosystem to get this done.
+
+## Demo
+
+But first, a very quick demo of how this looks like right now now!
+
+<!-- generated from original .mov screen recording with `ffmpeg -i screencast.mov -c:v libx264 screencast.mp4` -->
+<video src="./screencast.mp4"></video>
```

Comment:
@jnywong good call! I have fixed it now so it autoplays unless people have it blocked on their browser (like i do). I also added playback controls so it can be controlled.